spec/sec-properties-of-the-regexp-prototype-object-patch.html

```diff
@@ -139,10 +139,11 @@ <h1><ins>MakeIndicesArray ( _S_ , _indices_, _groupNames_, _hasGroups_ )</ins></
       <emu-alg>
         1. Assert: Type(_S_) is String.
         1. Assert: _indices_ is a List.
-        1. Assert: _groupNames_ is a List or is *undefined*.
         1. Assert: Type(_hasGroups_) is Boolean. 
         1. Let _n_ be the number of elements in _indices_.
         1. Assert: _n_ &lt; 2<sup>32</sup>-1.
+        1. Assert: _groupNames_ is a List with _n_ - 1 elements.
```

Comment:
just to confirm, there’s no way to get here when indices is empty?

Reply:
Yes. `indices` will always have at least one element in it (the whole pattern match), otherwise we would have returned `null` from RegExpBuiltinExec before reaching this point.